### DISCUSSION TITLE
Sonoff S26R2

### DISCUSSION THREAD

**Discussion by whopperg** [arendst/Tasmota#14064] [Q&A] [answered]
### PROBLEM DESCRIPTION
_Cannot flash Tasmota to new SONOFF S26R2._

### REQUESTED INFORMATION
- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)

- [X] Device used (e.g., Sonoff Basic): SONOFF S26R2
- [X] Tasmota binary firmware version number used: 10.1
  - [X] Pre-compiled

- [x] Flashing tools used: Tasmotizer and ESPHome-Flasher-1.4.0

**Hi everyone!**
I've got some of the [new S26R2](https://ewelinkcommunity.net/2021/07/itead-silenty-introduces-the-sonoff-s26r2-als-successor-to-the-sonoff-s26/?utm_source=rss&utm_medium=rss&utm_campaign=itead-silenty-introduces-the-sonoff-s26r2-als-successor-to-the-sonoff-s26) EU SmartPlugs Yesterday and tried to flash tasmota on it.

But i can't get tasmotizer 1.1a nor ESPHome Flasher 1.4.0 to recognize the ESP 8285 Module.
Here are some pictures of the new PCB and layout:

![20211216_084116](https://user-images.githubusercontent.com/7213981/146333043-fad2dcca-3147-4274-b6ef-1d9a6a5e861d.jpg)

![20211216_084138](https://user-images.githubusercontent.com/7213981/146333030-907d812f-3b8a-44d2-8e88-b96abbd78508.jpg)

![20211216_084205](https://user-images.githubusercontent.com/7213981/146333020-a1b89238-e31b-4f93-9bda-ac9234a6b15b.jpg)

![20211216_084223](https://user-images.githubusercontent.com/7213981/146333010-80055e36-738a-4495-bf5d-9a368600fe97.jpg)

![20211216_084736](https://user-images.githubusercontent.com/7213981/146332977-6de0580c-faf3-4580-abc3-68d832508512.jpg)

The system boots if connected to the USB to TTL adapter and the blue led is flashing (pairing mode), so power is there i guess.

If i put it into flashmode (holding the button while plug in the usb adapter, the led's stay off. Neither way the esp gets recognized.

So, has anyone another idea, or hint what i could do to flash tasmota to this device?

Best Regards and keep up the good work!!!

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by blakadder** [comment, marked as answer]
Did you ground GPIO0 to put it in flash mode?

**Reply by whopperg** [reply]
Hey, isnt this why i have to press the button? But i will give it a try...

**Reply by blakadder** [reply]
First you need to verify that the button is connected to GPIO0!

**Reply by whopperg** [reply]
Well therefor i need to know witch pin it is ;-) According to the [PINOUT](https://tasmota.github.io/docs/Pinouts/)  this should be 15. from the markings on the pcb it should be the 4th solderpoint from the left:

![20211216_120323](https://user-images.githubusercontent.com/7213981/146360556-b4a375de-846c-490c-a2f0-dae16100755c.jpg)

![20211216_120340](https://user-images.githubusercontent.com/7213981/146360560-ef829dba-f121-4c1b-ae98-75bff2ff434a.jpg)
But this does nothing. Blue led is flashing for pairing... if i also press the button, the blue led is off, but it also isnt recognized by ESPHome.

and unfortunally, i cannot solder directly to the chip because its way to small ;-)

**Reply by whopperg** [reply]
Found the correct pin (see Photo) but pulling to to GND not changing anything. Well I guess it's in flash mode because the blue LED is not flashing. 
I get a new USB-TTL adapter tomorrow to see if this makes any difference... (The current one still works and I flashed around 9ish S20 with it)

![20211216_131038](https://user-images.githubusercontent.com/7213981/146373401-2607ccae-73e4-41a4-b976-60e3ba21bd6b.jpg)


![20211216_131028](https://user-images.githubusercontent.com/7213981/146373385-0a48ebf2-7b96-4e2f-bacf-24f740e264dd.jpg)

**Reply by sfromis** [reply]
Are you using the actually recommended USB/Serial adapter? Some adapters seen much too often (especially fake "FTDI") are not that reliable for flashing due to not being good for powering a device while flashing.

**Reply by whopperg** [reply]
Hey sfromis,
I got these here. The red one did already flash 8to9 S20 successful...
![20211218_132542](https://user-images.githubusercontent.com/7213981/146641092-c71a7c88-2a1e-4eda-ae07-9bed7d9a54b0.jpg)

![20211218_132636](https://user-images.githubusercontent.com/7213981/146641094-6db74491-a5da-4f51-ab7c-4f13b41ffc6f.jpg)

**Reply by sfromis** [reply]
The golden CH340G is the better one (with 150 mA of 3.3V power), the red board is not recommended for reliable flashing, as it can't provide as much power.

**Reply by whopperg** [reply]
Yeah I know but thanks for confirming that. But also with the golden one, I don't have any luck. I will try a second s26r2, maybe I got a faulty one here...

**Reply by whopperg** [reply]
It is working now. **I'm stupid**. That's the reason why it didn't worked with the Gold USB TTL Adapter... I

Drum roll: I didn't switch Rx and Tx....
That's embarrassing...

Anyways: that's everybody how tried to help me!
I hope you all have marvelous holidays and stay negative! :-)

**Reply by sash85** [reply]
@whopperg Do I need to ground GPIO0 or will it also work without? I'm not very familiar with soldering and would like to avoid this step. :-)

**Reply by Andre1Becker** [reply]
Hey @sash85,
no, you don't need to do that. Just the 4 pins and your good to go :-)

**Reply by sfromis** [reply]
You do actually need to ground gpio0 before/while applying power for flashing, but that can be done simply by pushing the button on the device, no wiring needed.

**Reply by Andre1Becker** [reply]
Absolutely, I was referring to soldering, but this should make that clear. To enter flash mode, press the button as @sfromis  noted correctly :-)

**Comment by florin03** [comment]
hello,
Ignore blue led stats, that only works with sonoff app, you need to hold the device button pressed and connect usbttl to your PC, this will bring esp in flash mode, after that use any esp flasher, disconnect and reconnect usbttl to pc to reboot esp.
thx

**Comment by sash85** [comment]
Because I had some understanding problems, I wanted to share my guide.

1. Open the SONOFF S26R2 (three Screws) -> On the top ist a little plastic hook, thats why you should open this carefully.
2. Connect to the 4 Pins (GND to GND, 3V3 to V, TXD to RX and RXD to TX)
3. Push and hold the Button (GPIO 0) **and now** plugin the usb
4. Hold the Button all the time and install Tasmota (e.g. with webinstaller: https://tasmota.github.io/install/)
5. After the install is complete: Release the pushbutton and plugout the usb cable
6. Plugin in the usb cable again and connect again with the web installer (Connect -> choose "USB...").
7. At this point you should see the option "CONNECT TO WI-FI" (Sometimes it can happen, that you only see the install option -> press install, get over the error and after that you should see the install Option)
8. After connecting to your wifi, everthing should work fine. :-) 

At first is solder the pins but i doesn't want to solder every Device. Thats why i used another solution:
![IMG_8536](https://user-images.githubusercontent.com/10808627/173018734-49ee4fcc-864a-4ed2-a1c9-6d40e8d120f4.jpeg)

**Comment by corradodave** [comment]


**Reply by Andre1Becker** [reply]
Hey, try to press it a little harder. Some Buttons are a bit stiffer from my expirience

**Comment by Gajusbonus** [comment]
![dialout-group](https://github.com/arendst/Tasmota/assets/7600670/b396c612-39c4-4666-b31f-96a5f8fcecf3)
And don't forget to put yourself in the dialoutgroup !

**Comment by obbardc** [comment]
I found these devices (I have 4) buzzes quite bad with ESPhome flashed and the relay on. Does it happen to anyone else ?

**Reply by sfromis** [reply]
Buzzing is often a sign of senescent devices with worn down caps.

**Reply by obbardc** [reply]
Right - these were buzzing since new (~2020) and I put them in storage since then. I got them out yesterday and updated esphome, same thing..

**Reply by sfromis** [reply]
Updating caps has a better chance of an effect.
